drivers/SmartThings/philips-hue/src/disco/light.lua

```diff
@@ -29,7 +29,13 @@ local function join_light(driver, light, device_service_info, parent_device_id,
     return
   end
 
-  local device_name = light.metadata.name
+  local device_name
+  if light.metadata.name == device_service_info.metadata.name then
```

Comment:
Im not sure if nil checks are warranted here. Indexing like this makes me concerned when they aren't present, but maybe things are guaranteed non-nil?

Reply:
`metadata` is guaranteed on all services.
`metadata.name` is guaranteed on all `device` services.
`metadata.name` is not guaranteed on all other services, but it *is* guaranteed on `light` services. We confirmed this with Signify.